### ISSUE TITLE
5.x newtonsoft.json version 12

### ISSUE THREAD

**Issue by gautelo** [elastic/elasticsearch-net#3501]
newtonsoft.json has released a version 12. Unfortunately nest/elasticsearch 5.x require that newtonsoft.json is lower than version 12 for some reason. This causes incompatibility issues with packages which require version 12 or higher newtonsoft.json.

Please lift this restriction! Thanks.

**Comment by Mpdreamz** [comment]
Hi! We'll run our tests with 12 and release an update accordingly as soon as we can.

The restriction is in place because major versions have introduced breaking changes (as major versions are totally entitled too) in the past.

Although i did remove the upper limit in https://github.com/elastic/elasticsearch-net/pull/3127 in `6.x` and `master` for `Nest.JsonNetSerializer`. I am tempted to do the same for `5.x` but would like a quorum on that decision first @russcam @codebrain

**Comment by gautelo** [comment]
Thanks. Sounds great! :) That's a reasonable position.

In support of lifting the upper limit on Newtonsoft.Json, I've got to say that even though you may have had issues with not locking Newtonsoft.Json to a specific version in the past, my impression is that the library is very stable. Furthermore I know microsoft is developing their own new faster json library. The author of Newtonsoft.Json did not feel like he could optimize Newtonsoft.Json the way they wanted/needed because it would cause too much breakage. After all, there is a LOT of code depending on that library.

**Comment by Mpdreamz** [comment]
Opened a PR for this #3507, we should release this next week. Thanks for raising this @gautelo

**Comment by gautelo** [comment]
Sorry to be an impatient pest, but when can we expect this to be released? I'm kind of eager to get rid of all of my redundant and ugly no-warn references :)

```
    <PackageReference Include="Newtonsoft.Json" Version="12.0.1">
      <NoWarn>NU1608</NoWarn>
    </PackageReference>
```

**Comment by Mpdreamz** [comment]
We just released [6.4.1 and 5.6.5](https://github.com/elastic/elasticsearch-net/releases) that bump the `Newtonsoft.Json` dependency to `12`. It might take a bit for them to show up on nuget.org.

Totally appreciate the bump @gautelo 👍 Was pulled in 100 different directions this week so was going to hold off till Monday.
